provider/bidengine/order_test.go

```diff
@@ -0,0 +1,591 @@
+func (_ fauxQueryClient) Orders(ctx context.Context, in *mtypes.QueryOrdersRequest, opts ...grpc.CallOption) (*mtypes.QueryOrdersResponse, error) {
+	// Return no orders
+	return &mtypes.QueryOrdersResponse{}, nil
+}
+
+// Order queries order details
+func (_ fauxQueryClient) Order(ctx context.Context, in *mtypes.QueryOrderRequest, opts ...grpc.CallOption) (*mtypes.QueryOrderResponse, error) {
+	return nil, errNotImplemented
+}
+
+// Bids queries bids with filters
+func (_ fauxQueryClient) Bids(ctx context.Context, in *mtypes.QueryBidsRequest, opts ...grpc.CallOption) (*mtypes.QueryBidsResponse, error) {
+	return nil, errNotImplemented
+}
+
+// Bid queries bid details
+func (_ fauxQueryClient) Bid(ctx context.Context, in *mtypes.QueryBidRequest, opts ...grpc.CallOption) (*mtypes.QueryBidResponse, error) {
+	return nil, errNotImplemented
+}
+
+// Leases queries leases with filters
+func (_ fauxQueryClient) Leases(ctx context.Context, in *mtypes.QueryLeasesRequest, opts ...grpc.CallOption) (*mtypes.QueryLeasesResponse, error) {
+	return nil, errNotImplemented
+}
+
+// Lease queries lease details
+func (_ fauxQueryClient) Lease(ctx context.Context, in *mtypes.QueryLeaseRequest, opts ...grpc.CallOption) (*mtypes.QueryLeaseResponse, error) {
+	return nil, errNotImplemented
+}
+
+func (_ fauxQueryClient) ActiveLeasesForProvider(id sdk.AccAddress) (mtypes.Leases, error) {
+	return nil, errNotImplemented
+}
+
+type fauxTxClient struct {
+	broadcasts chan []sdk.Msg
+}
+
+func newFauxTxClient() fauxTxClient {
+	return fauxTxClient{
+		broadcasts: make(chan []sdk.Msg),
+	}
+}
+
+func (ftc fauxTxClient) Broadcast(msgs ...sdk.Msg) error {
+	ftc.broadcasts <- msgs
+	return nil
+}
+
+type fauxClient struct {
+	ftc fauxTxClient
+}
+
+func newFauxClient() fauxClient {
+	return fauxClient{
+		ftc: newFauxTxClient(),
+	}
+}
+
+func (_ fauxClient) Query() client.QueryClient {
+	return fauxQueryClient(0)
+}
+
+func (fc fauxClient) Tx() client.TxClient {
+	return fc.ftc
+}
+
+type fauxReservation struct {
+	orderId   mtypes.OrderID
+	resources atypes.ResourceGroup
+}
+
+func (fr fauxReservation) OrderID() mtypes.OrderID {
+	return fr.orderId
+}
+
+func (fr fauxReservation) Resources() atypes.ResourceGroup {
+	return fr.resources
+}
+
+type fauxCluster struct {
+	reserveCallCount int
+	lastReserveCall  struct {
+		orderId       mtypes.OrderID
+		resourceGroup atypes.ResourceGroup
+	}
+	reserveCallNotify chan int
+
+	unreserveCallCount int
+	lastUnreserveCall  struct {
+		orderId       mtypes.OrderID
+		resourceGroup atypes.ResourceGroup
+	}
+}
+
+func (fc *fauxCluster) Reserve(orderID mtypes.OrderID, rg atypes.ResourceGroup) (cluster.Reservation, error) {
+	fc.reserveCallCount++
+	fmt.Printf("Reserve call order ID OSeq: %d\n", orderID.OSeq)
+	lastCall := &fc.lastReserveCall
+	lastCall.resourceGroup = rg
+	lastCall.orderId = orderID
+
+	select {
+	case fc.reserveCallNotify <- 0:
+
+	default:
+	}
+
+	return fauxReservation{orderId: orderID, resources: rg}, nil
```

Comment:
I feel like we should just make [this](https://github.com/ovrclk/akash/blob/494572ac32a7a48d801a27397e6d2047d80419ec/provider/cluster/reservation.go#L14) public.

Reply:
That's possible but what would the exported thing be named? We can't use `Reservation` since the interface is already called that. We'd have to use some name like `ReservationImpl` , `BasicReservation` or `ConcreteReservation`

Reply:
I am using a mock for this now